src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -419,64 +419,228 @@ private void LookupMetadataDefinitions(
             TypeDefinition typeDefinition,
             OrderPreservingMultiDictionary<string, MetadataDefinition> definitionMap)
         {
-            // Only bother looking for extension methods in static types.
-            // Note this check means we would ignore extension methods declared in assemblies
-            // compiled from VB code, since a module in VB is compiled into class with 
-            // "sealed" attribute but not "abstract". 
-            // Although this can be addressed by checking custom attributes,
-            // we believe this is not a common scenario to warrant potential perf impact.
-            if ((typeDefinition.Attributes & TypeAttributes.Abstract) != 0 &&
-                (typeDefinition.Attributes & TypeAttributes.Sealed) != 0)
+            var currentTypeIsStaticClass = IsStaticClass(typeDefinition);
+
+            // Only bother looking for extension methods in static types. Note this check means we would ignore
+            // extension methods declared in assemblies compiled from VB code, since a module in VB is compiled into
+            // class with "sealed" attribute but not "abstract". Although this can be addressed by checking custom
+            // attributes, we believe this is not a common scenario to warrant potential perf impact.
+            if (currentTypeIsStaticClass)
+            {
+                // Handle classic extension methods.
+                _containsExtensionMembers = LoadClassicExtensionMethods() || _containsExtensionMembers;
+            }
+
+            foreach (var child in typeDefinition.GetNestedTypes())
+            {
+                var nestedType = metadataReader.GetTypeDefinition(child);
+
+                // We don't include internals from metadata assemblies.  It's less likely that a project would have IVT
+                // to it and so it helps us save on memory.  It also means we can avoid loading lots and lots of
+                // obfuscated code in the case the dll was obfuscated.
+                if (IsPublicType(nestedType))
```

Comment:
Could also check `IsExtension` to confirm this is an extension block

Reply:
not sure how to do that.  What let's know an extensino block is an extension block?

Reply:
My comment is not applicable. We're dealing with metadata directly (not symbols), so `IsExtension` isn't available